test/nut/sandboxCreate.nut.ts

```diff
@@ -0,0 +1,166 @@
+import { SandboxProcessObject, SandboxRequestCache } from '@salesforce/core';
+import { stubSfCommandUx, stubSpinner, stubUx } from '@salesforce/sf-plugins-core';
+import CreateSandbox from '../../src/commands/org/create/sandbox.js';
+import {
+  deleteSandboxCacheFile,
+  getSandboxInfo,
+  getSandboxProcess,
+  getSandboxProcessSoql,
+  readSandboxCacheFile,
+  stubProdOrgConnection,
+  stubToolingCreate,
+  stubToolingQuery,
+} from '../shared/sandboxMockUtils.js';
+
+describe('Sandbox Create', () => {
+  let session: TestSession;
+  let hubOrgUsername: string;
+  let cacheFilePath: string;
+  let sandboxDefFilePath: string;
+  let sfCommandUxStubs: ReturnType<typeof stubSfCommandUx>;
+
+  const sinonSandbox = sinon.createSandbox();
+
+  let sandboxProcessSoql: string;
+
+  before(async () => {
+    const uid = genUniqueString('sbxCreate_%s');
+    session = await TestSession.create({
+      project: { name: 'sandboxCreate' },
+      devhubAuthStrategy: 'AUTH_URL',
+      sessionDir: path.join(process.cwd(), `test_session_${uid}`),
+    });
+    assert(session.hubOrg.username);
+    hubOrgUsername = session.hubOrg.username;
+    cacheFilePath = path.join(session.dir, '.sf', SandboxRequestCache.getFileName());
+    sandboxDefFilePath = path.join(session.project.dir, 'sandboxDef.json');
+
+    // add a sandbox definition file to the project
+    const { SandboxName, LicenseType, Id } = getSandboxInfo();
+    sandboxProcessSoql = getSandboxProcessSoql({ sandboxInfoId: Id });
+    fs.writeFileSync(sandboxDefFilePath, JSON.stringify({ SandboxName, LicenseType }));
+  });
+
+  beforeEach(() => {
+    sfCommandUxStubs = stubSfCommandUx(sinonSandbox);
+    stubUx(sinonSandbox);
+    stubSpinner(sinonSandbox);
+  });
+
+  after(async () => {
+    await session?.clean();
+  });
+
+  afterEach(() => {
+    try {
+      deleteSandboxCacheFile(cacheFilePath);
+    } catch (err) {
+      // ignore since there isn't always a cache file written
+    }
+    sinonSandbox.restore();
+  });
+
+  //
+  // NOTE: These tests use stubbed server responses since sandbox refresh
+  //       takes a very long time.
+  // Stubs:
+  //   1. Connection.singleRecordQuery - used to query for the SandboxInfo
+  //   2. Connection.tooling.update - used to update the SandboxInfo
+  //   3. Connection.tooling.query - used to query for the SandboxProcess
+  //
+
+  it('should return a SandboxProcessObject without polling (--async) using --name', async () => {
+    const sbxInfo = getSandboxInfo();
+    const sbxName = sbxInfo.SandboxName;
+    const sbxLicenseType = 'Developer';
+    const sbxProcess = getSandboxProcess();
+    const connection = await stubProdOrgConnection(sinonSandbox, hubOrgUsername);
+
+    const toolingCreateStub = stubToolingCreate({ sinonSandbox, connection });
+    const toolingQueryStub = stubToolingQuery({ sinonSandbox, connection, sandboxProcessSoql, sbxProcess });
+
+    const result: SandboxProcessObject = await CreateSandbox.run([
```

Comment:
you shouldn't need the `: SandboxProcessObject` here (or anywhere you're using the static `run` method on `Command`)